navigation2menu.js

```diff
@@ -18,7 +18,8 @@ for (const line of markdown.split('\n')) {
         continue;
     }
     const indentPart = line.replace(/(^ *\* ).*/, '$1');
-    const markdownLink = line.substring(indentPart.length);
+    const markdownLink = line.substring(indentPart.length)
+         .replace(/#.*$/, '');
     const matched = markdownLink.match(/\[([^\]]+)\]\(([^)]+)\)/);
```

Comment:
Btw., just a suggestion, but maybe we could extract this into a new function and add some tests for it? I've had a hard time with regex lately, and it would be great to have some tests as a safety net if we need to make changes in the future.